src/librbd/AioRequest.cc

```diff
@@ -201,13 +201,10 @@ namespace librbd {
     ldout(m_ictx->cct, 20) << "send " << this << " " << m_oid << " " << m_object_off << "~" << m_object_len << dendl;
 
     // send read request to parent if the object doesn't exist locally
-    {
-      RWLock::RLocker l(m_ictx->md_lock);
-      if (m_ictx->object_map != NULL &&
-	  !m_ictx->object_map->object_may_exist(m_object_no)) {
-	complete(-ENOENT);
-	return 0;
-      }
+    if (m_ictx->object_map_enabled() &&
```

Comment:
So the first condition is protected by a lock and the next one isn't.  This issue is throughout.